src/main/java/org/mvel2/execution/ExecutionArrayList.java

```diff
@@ -116,13 +165,134 @@ public long memorySize() {
-                .map( Object::toString )
-                .collect( Collectors.joining( separator ) );
+        List rez = new ArrayList<String>();
+        this.stream().forEach(e -> {
+                    if (e instanceof List) {
+                        rez.add(((List) e).stream().map(Object::toString).collect(Collectors.joining(separator)));
+                    } else {
+                        rez.add(e.toString());
+                    }
+                }
+        );
+        return (String) rez.stream().collect(Collectors.joining(separator));
+    }
+
+    public void sort() {
+        this.sort(true);
+    }
+
+    public void sort(boolean asc) {
+        if (validateClazzInArrayIsOnlyString()) {
+            super.sort(asc ? null : stringCompDesc);
+        } else {
+            super.sort(asc ? numericCompAsc : numericCompDesc);
+        }
+    }
+
+    public ExecutionArrayList<E> toSorted() {
+        return this.toSorted(true);
+    }
+
+    public ExecutionArrayList<E> toSorted(boolean asc) {
+        ExecutionArrayList newList = this.slice();
+        newList.sort(asc);
+        return newList;
+    }
+
+    public void reversed() {
+        Collections.reverse(this);
+    }
+
+    public List toReversed() {
+        ExecutionArrayList newList = this.slice();
+        newList.reversed();
+        return newList;
+    }
+
+    public List concat(Collection c) {
+        ExecutionArrayList newList = this.slice();
+        newList.addAll(c);
+        return newList;
+    }
+
+    public List splice(int start) {
+        return this.splice(start, null);
+    }
+
+    public List splice(Object st, Object delCount, E... items) {
+        int start = Integer.parseInt(String.valueOf(st));
+        start = start < -this.size() ? 0 : start >= this.size() ? this.size() : start < 0 ? start + this.size() : start;
+        int deleteCount = delCount == null ? this.size() - start : Integer.parseInt(String.valueOf(delCount));
+        deleteCount = start == this.size() ? 0 : Math.max(deleteCount, 0);
+        List<E> removed = new ArrayList<>();
+        int deleteIdx = deleteCount == 0 ? this.size() : start;
+        AtomicInteger insertIdx = new AtomicInteger(start);
+        while (deleteCount > 0) {
+            removed.add(this.remove(deleteIdx));
+            deleteCount--;
+        }
+        for (E e : items) {
+            this.add(insertIdx.getAndIncrement(), e);
+        }
+        return new ExecutionArrayList<>(removed, this.executionContext);
+    }
+
+    public List toSpliced(int start) {
+        return this.toSpliced(start, null);
+    }
+
+    public List toSpliced(Object st, Object delCount, E... items) {
+        ExecutionArrayList newList = this.slice();
+        newList.splice(st, delCount, items);
+        return newList;
+    }
+
+    public List with(Object ind, E items) {
+        int index = Integer.parseInt(String.valueOf(ind));
+        if (index >= this.size() || index < -this.size()) {
```

Comment:
Improve implementation. 
1. First you should calculate the real index if it is negative. 
2. Then you should check that the index is in bounds.
3. Then create new array and change the value of the element by index.

Reply:
Ok